From: Hannes Reinecke <hare@suse.de>
Date: Fri, 22 Mar 2019 11:41:57 +0100
Subject: [PATCH] drivers/base: kABI fixes for struct device_private
References: bsc#1106383
Patch-Mainline: never, kABI fix for SLE15

kABI fixes for updates in device_private.
Should not be necessary, but our kABI checker insists...

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/base/base.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/base/base.h b/drivers/base/base.h
index bef4256f8bca..9337cd4c31bf 100644
--- a/drivers/base/base.h
+++ b/drivers/base/base.h
@@ -78,9 +78,11 @@ struct device_private {
 	struct klist_node knode_driver;
 	struct klist_node knode_bus;
 	struct list_head deferred_probe;
-	struct device_driver *async_driver;
 	struct device *device;
+#ifndef __GENKSYMS__
+	struct device_driver *async_driver;
 	u8 dead:1;
+#endif
 };
 #define to_device_private_parent(obj)	\
 	container_of(obj, struct device_private, knode_parent)
-- 
2.16.4

